airflow/utils/log/stackdriver_task_handler.py

```diff
@@ -0,0 +1,286 @@
+                     to the global resource type.
+    :type resource: :class:`~google.cloud.logging.resource.Resource`
+    :param labels: (Optional) Mapping of labels for the entry.
+    :type labels: dict
+    """
+
+    LABEL_TASK_ID = "task_id"
+    LABEL_DAG_ID = "dag_id"
+    LABEL_EXECUTION_DATE = "execution_date"
+    LABEL_TRY_NUMBER = "try_number"
+
+    def __init__(
+        self,
+        gcp_conn_id: Optional[str] = None,
+        name: str = DEFAULT_LOGGER_NAME,
+        transport: Type[Transport] = BackgroundThreadTransport,
+        resource: Resource = _GLOBAL_RESOURCE,
+        labels: Optional[Dict[str, str]] = None,
+    ):
+        super().__init__()
+        self.gcp_conn_id = gcp_conn_id
+        self.name: str = name
+        self.transport_type: Type[Transport] = transport
+        self.resource: Resource = resource
+        self.labels: Optional[Dict[str, str]] = labels
+        self.task_instance_labels: Optional[Dict[str, str]] = {}
+
+    @cached_property
+    def _client(self) -> gcp_logging.Client:
+        """Google Cloud Library API clinet"""
+        if self.gcp_conn_id:
+            from airflow.gcp.hooks.base import GoogleCloudBaseHook
+
+            hook = GoogleCloudBaseHook(gcp_conn_id=self.gcp_conn_id)
+            credentials = hook._get_credentials()  # pylint: disable=protected-access
+        else:
+            # Use Application Default Credentials
+            credentials = None
+        client = gcp_logging.Client(
+            credentials=credentials,
+            client_info=ClientInfo(client_library_version='airflow_v' + version.version)
+        )
+        return client
+
+    @cached_property
+    def _transport(self) -> Transport:
+        """Object responsible for sending data to Stackdriver"""
+        return self.transport_type(self._client, self.name)
+
+    def emit(self, record: logging.LogRecord) -> None:
+        """Actually log the specified logging record.
+
+        :param record: The record to be logged.
+        :type record: logging.LogRecord
+        """
+        message = self.format(record)
+        labels: Optional[Dict[str, str]]
+        if self.labels and self.task_instance_labels:
+            labels = {}
+            labels.update(self.labels)
+            labels.update(self.task_instance_labels)
+        elif self.labels:
+            labels = self.labels
+        elif self.task_instance_labels:
+            labels = self.task_instance_labels
+        else:
+            labels = None
+        self._transport.send(record, message, resource=self.resource, labels=labels)
+
+    def set_context(self, task_instance: TaskInstance) -> None:
+        """
+        Configures the logger to add information with information about the current task
+
+        :param task_instance: Currently executed task
+        :type task_instance: TaskInstance
+        """
+        self.task_instance_labels = self._task_instance_to_labels(task_instance)
+
+    def read(
```

Comment:
I would like to avoid modifying the API, but in this change I would like to implement the current Airflow interface. Currently in Airflow the process of saving and reading is combined in one class.  
If you want, you can configure another handler to read data using ``task_log_reader `` option in ``airflow.cfg``.
https://github.com/apache/airflow/blob/master/airflow/config_templates/default_airflow.cfg#L191-L193
I agree that the log API needs improvement, but this is not the scope of this PR.

Can you look at my comment that discusses API issues?
https://github.com/apache/airflow/pull/5177#issuecomment-557788417